### PR TITLE
FIX | TestMailer uses new testsession APIs for handling state

### PR DESCRIPTION
**Pull request by dnsl48**
This PR is a follow up on https://github.com/silverstripe/silverstripe-testsession/pull/65

### CODE REVIEW

File: src/Utility/TestMailer.php [hunk 1]
```diff
@@ -94,6 +94,6 @@ protected function saveEmail($data)
             $state->emails = array();
         }
         $state->emails[] = array_filter($data);
-        $this->testSessionEnvironment->applyState($state);
+        $this->testSessionEnvironment->saveState($state);
```
Comment (robbieaverill) [inline src/Utility/TestMailer.php]: The change looks OK. The doc blocks aren't particularly useful and I'm not familiar with the difference, but presumably this has been tested in behat and fixes a problem, so LGTM

**Comment by robbieaverill** [comment]
Would you mind fixing the builds to ensure they're still green?

This branch is for SilverStripe 5.x, so you may want to retarget the PR. The Travis config needs recipe constraints changed from 2.x to 5.x

**Comment by dnsl48** [comment]
sure, will do, thank you for taking a look :)

**Comment by GuySartorelli** [comment]
@dnsl48 Do you know what specifically this was solving? Was there a bug that this fixes? I'm inclined to close this PR given we've been running behat tests for 4+ years since the PR was opened and this hasn't seemed to be a problem.

**Comment by dnsl48** [comment]
Hey @GuySartorelli, sorry, I don't remember the exact details.
Looks like everything is okay as is without this patch :)

**Pull request closed without merging**